assets/app/components/site/siteBuildLogs.js

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import buildLogActions from "../../actions/buildLogActions";
+
+class SiteBuildLogs extends React.Component {
+  componentDidMount() {
```

Comment:
"Something changed! We need to show the site build logs!"

At the very moment when we know we're going to want to have a siteBuildLogs component, we also know we want to fetch the data needed by that component. We should fetch the data then instead of relying on the React lifecycle.

If we know enough to change the redux state to signify we're waiting on data, then we know enough to trigger the fetch whose response will populate the redux state through a state transition. The closer we can align the causality of these two events (the redux "pending" transition and the data fetch), the easier it'll be to follow the dependency.

If the fetching bit is triggered in a "container" component whose only purpose is to trigger the fetch of the data, then we're looking at a path where we...
  * Alter the redux state to put us in the "we're about to care about the build logs" state
  * A particular container component is instantiated by React as the state is passed down the render tree
  * React calls componentWillMount on the container component
  * Fetch happens
  * Response triggers the Redux state transition with the new data

Is the container responsible for holding the components that will render when the data to be fetched? That's just one more thing for it to be responsible for.

Does it really need to be that complicated?

If we trigger the fetch at the moment when we know we need the data, our path becomes
  * Alter the redux state to put us in the "we're about to care about the build logs" state
  * Fetch happens
  * Response triggers the Redux state transition with the new data

Then nothing depends on the Redux component lifecycle. Components are simpler and easier to test. Containers can have the singular responsibility of bridging the Redux state into the React props tree.

I'd use the React lifecycle to trigger things that really really really depend on that lifecycle, e.g. "we need to wait to run this code until after we're certain our components have mounted in the DOM."